File: Serene/Serene.Script/Northwind/Shipper/ShipperGrid.cs

namespace Serene.Northwind
{
    using jQueryApi;
    using Serenity;
    using System.Runtime.CompilerServices;

    [Imported]
    [ColumnsKey("Northwind.Shipper"), IdProperty("ShipperID"), NameProperty("CompanyName")]
    [DialogType(typeof(ShipperDialog)), LocalTextPrefix("Northwind.Shipper"), Service("Northwind/Shipper")]
    public class ShipperGrid : EntityGrid<ShipperRow>
    {
        public ShipperGrid(jQueryObject container)
            : base(container)
        {
        }
    }
}

File: Serene/Serene.Core/Modules/Northwind/Shipper/ShipperGrid.ts
namespace Serene.Northwind {

    @Serenity.Decorators.registerClass()
    export class ShipperGrid extends Serenity.EntityGrid<ShipperRow, any> {
        protected getColumnsKey() { return "Northwind.Shipper"; }
        protected getDialogType() { return <any>ShipperDialog; }
        protected getIdProperty() { return ShipperRow.idProperty; }
        protected getLocalTextPrefix() { return ShipperRow.localTextPrefix; }
        protected getService() { return ShipperService.baseUrl; }

        constructor(container: JQuery) {
            super(container);
        }
    }
}